packages/edit-navigation/src/store/controls.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * WordPress dependencies
+ */
+import { default as triggerApiFetch } from '@wordpress/api-fetch';
+import { createRegistryControl } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { menuItemsQuery } from './utils';
+
+/**
+ * Trigger an API Fetch request.
+ *
+ * @param {Object} request API Fetch Request Object.
+ * @return {Object} control descriptor.
+ */
+export function apiFetch( request ) {
+	return {
+		type: 'API_FETCH',
+		request,
+	};
+}
+
+export function getPendingActions( postId ) {
+	return {
+		type: 'GET_PENDING_ACTIONS',
+		postId,
+	};
+}
+
+export function isProcessingPost( postId ) {
+	return {
+		type: 'IS_PROCESSING_POST',
+		postId,
+	};
+}
+
+export function getMenuItemToClientIdMapping( postId ) {
+	return {
+		type: 'GET_MENU_ITEM_TO_CLIENT_ID_MAPPING',
+		postId,
+	};
+}
```

Comment:
I ended up moving this mapping back to state - keeping it in the post turned out to be troublesome (e.g. every update created additional undo actions). I'm not sure if there is any other way to have a "private selector" other than custom control - I'm open to refactoring this bit.